catalog/spatial/registry/registry-federation-admin-service-impl/src/main/java/org/codice/ddf/registry/federationadmin/service/impl/IdentityNodeInitialization.java

```diff
@@ -85,25 +87,35 @@
 
   private long retryInterval = DEFAULT_RETRY_INTERVAL;
 
-  private int retryCount = DEFAULT_RETRY_COUNT;
+  private ScheduledExecutorService executor;
 
-  public IdentityNodeInitialization() {}
+  public IdentityNodeInitialization(ScheduledExecutorService executor) {
+    this(executor, DEFAULT_RETRY_INTERVAL);
+  }
 
-  public IdentityNodeInitialization(long retryInterval, int retryCount) {
+  public IdentityNodeInitialization(ScheduledExecutorService executor, long retryInterval) {
     this.retryInterval = retryInterval;
-    this.retryCount = retryCount;
+    this.executor = executor;
   }
 
-  public void init() throws InterruptedException {
+  public void init() {
+    // will keep checking for changes and shouldn't hold up anything
+    executor.scheduleAtFixedRate(this::updateOrCreateIdentity, 1, retryInterval, TimeUnit.SECONDS);
+  }
 
-    for (int count = 0; count < retryCount; count++) {
-      if (updateOrCreateIdentity()) {
-        return;
+  public void destroy() {
+    executor.shutdown();
+    try {
+      if (!executor.awaitTermination(SHUTDOWN_TIMEOUT_SECONDS, TimeUnit.SECONDS)) {
+        executor.shutdownNow();
+        if (!executor.awaitTermination(SHUTDOWN_TIMEOUT_SECONDS, TimeUnit.SECONDS)) {
+          LOGGER.error("Thread pool didn't terminate");
```

Comment:
✏️ Should this be logged at a lower level instead? Would it be helpful to add a message telling a system admin what the consequences are and how to recover? https://codice.atlassian.net/wiki/spaces/DDF/pages/7045131/Log+Levels

Reply:
Updated message